apisix/plugins/cors.lua

```diff
@@ -37,7 +38,7 @@ local metadata_schema = {
             type = "object",
             additionalProperties = {
                 type = "string",
-                pattern = origins_pattern
+                pattern = origins_metadata_pattern
```

Comment:
Why change this pattern?

Reply:
Because else the tests for the `metadata_schema` started failing and I don't understand what `metadata_schema` is used for so I wanted to save and not sorry.

However if you are fine saying that an empty string is a valid origin we can avoid this and I can fix the text.

Reply:
an empty string is not a valid origin. We can keep the same regex in metadata_schema and the others.

Reply:
I think this is incorrect as the allow_origins can be empty or unspecified if a regex is allow_origins_by_regex is specified.

However I didn't find any way to make `allow_origins` optional as it has a default value of `*`.

Reply:
I notice there is a fix recently: https://github.com/apache/apisix/pull/9028